openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -496,3 +544,165 @@ class CourseOverviewTab(models.Model):
+       mean that any thumbnails exist. There might have been a processing error,
+       or there might simply be no source image to create a thumbnail out of.
+       In this case, accessing CourseOverview.image_urls will return the value
+       for course.course_image_url for all resolutions. CourseOverviewImageSet
+       will *not* try to regenerate if there is a model entry with blank values
+       for the URLs -- the assumption is that either there's no data there or
+       something has gone wrong and needs fixing in code.
+
+    5. If you want to change thumbnail resolutions, you need to create a new
+       CourseOverviewImageConfig with the desired dimensions and then wipe the
+       values in CourseOverviewImageSet.
+
+    Logical next steps that I punted on for this first cut:
+
+    1. Center cropping the image before scaling.
+
+       This is desirable, but it involves a few edge cases (what the rounding
+       policy is, what to do with undersized images, etc.) The behavior that
+       we implemented is at least no worse than what was already there in terms
+       of distorting images.
+
+    2. Automatically invalidating entries based on CourseOverviewImageConfig.
+
+       There are two basic paths I can think of for this. The first is to
+       completely wipe this table when the config changes. The second is to
+       actually tie the config as a foreign key from this model -- so you could
+       do the comparison to see if the image_set's config_id matched
+       CourseOverviewImageConfig.current() and invalidate it if they didn't
+       match. I punted on this mostly because it's just not something that
+       happens much at all in practice, there is an understood (if manual)
+       process to do it, and it can happen in a follow-on PR if anyone is
+       interested in extending this functionality.
+
+    """
+    course_overview = models.OneToOneField(CourseOverview, db_index=True, related_name="image_set")
+    small_url = models.TextField(blank=True, default="")
+    large_url = models.TextField(blank=True, default="")
+
+    @classmethod
+    def create_for_course(cls, course_overview, course=None):
+        """
+        Create thumbnail images for this CourseOverview.
+
+        This will save the CourseOverviewImageSet it creates before it returns.
+        """
+        from openedx.core.lib.courses import create_course_image_thumbnail
+
+        # If image thumbnails are not enabled, do nothing.
+        config = CourseOverviewImageConfig.current()
+        if not config.enabled:
+            return
+
+        # If a course object was provided, use that. Otherwise, pull it from
+        # CourseOverview's course_id. This happens because sometimes we are
+        # generated as part of the CourseOverview creation (course is available
+        # and passed in), and sometimes the CourseOverview already exists.
+        if not course:
+            course = modulestore().get_course(course_overview.id)
```

Comment:
True, but in practice I suspect that we'll bump the CourseOverview version more frequently than the thumbnail configuration. I mean, we haven't really changed the sizes we display on the dashboard or about page since 2012.

I have one possible bug I'm tracking down, and then I'll look for a cheap alternative. I don't really like adding `course_image` to the model because I feel like it would be more confusing for people.